components/analytics-solutions-common/org.wso2.analytics.solutions.common/widgets/DateTimeRangePicker/src/DateTimePicker.jsx

```diff
@@ -78,49 +78,125 @@ export default class DateTimePicker extends React.Component {
         } else {
             days = 30;
         }
-
-        for (let i = 1; i <= days; i++) {
-            dayComponents.push(
-                <MenuItem
-                    key={`$days-${i}`}
-                    value={i}
-                    primaryText={i}
-                />
-            );
+        if(inputName === 'startTime') {
+            for (let i = 1; i <= days; i++) {
+                dayComponents.push(
+                    <MenuItem
+                        key={`$days-${i}`}
+                        value={i}
+                        primaryText={i}
+                    />
+                );
+            }
+        } else if(inputName === 'endTime') {
+            if(moment(startTime).month() === this.state.month) {
+                const startDate = moment(startTime).date();
+                for (let i = startDate; i <= days; i++) {
+                    dayComponents.push(
+                        <MenuItem
+                            key={`$days-${i}`}
+                            value={i}
+                            primaryText={i}
+                        />
+                    );
+                }
+                if(this.state.days < startDate) {
+                    this.setState({ days: startDate + 1 });
+                    this.handleOnChange('days', startDate + 1 )
+                }
+            } else {
+                for (let i = 1; i <= days; i++) {
+                    dayComponents.push(
+                        <MenuItem
+                            key={`$days-${i}`}
+                            value={i}
+                            primaryText={i}
+                        />
+                    );
+                }
+            }
         }
 
         return dayComponents;
     }
 
-    generateMonths() {
+    generateMonths(inputName, startTime) {
         const monthComponents = [];
         const monthArray = ['January', 'February', 'March', 'April', 'May', 'June', 'July', 'August', 'September',
             'October', 'November', 'December'];
 
-        for (let i = 0; i < monthArray.length; i++) {
-            monthComponents.push(
-                <MenuItem
-                    key={`month-${i}`}
-                    value={i}
-                    primaryText={monthArray[i]}
-                />
-            );
+        if(inputName === 'startTime') {
+            for (let i = 0; i < monthArray.length; i++) {
+                monthComponents.push(
+                    <MenuItem
+                        key={`month-${i}`}
+                        value={i}
+                        primaryText={monthArray[i]}
+                    />
+                );
+            }
+        } else if(inputName === 'endTime') {
+            const start = moment(startTime);
+            const yearDiff = this.state.year-start.year();
```

Comment:
we can use de structuring for this.state.year

Reply:
Fixed with https://github.com/wso2/analytics-solutions/pull/306